### PR TITLE
docs: Updated Node.js get started guide

### PR DESCRIPTION
**Pull request by vikram-dagger**
This commit updates the Node.js get started guide to include additional TypeScript packages for new users unfamiliar with configuring Node.js' TypeScript support.

### CODE REVIEW

**Comment by vikram-dagger** [comment]
Ping @slumbering @crjm for review

**Review by crjm — COMMENTED** [review]
I tried to follow the guide in Typescript from [here](https://deploy-preview-5030--devel-docs-dagger-io.netlify.app/sdk/nodejs/783645/get-started).

When I do `node --loader ts-node/esm ./build.mts` (Step 2) to execute the script I get the following error:
```console
❯ node --loader ts-node/esm ./build.mts
(node:18625) ExperimentalWarning: Custom ESM Loaders is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
ReferenceError: exports is not defined in ES module scope
    at file:///tmp/dagger-node-test/build.mts:2:23
    at ModuleJob.run (node:internal/modules/esm/module_job:193:25)
    at async Promise.all (index 0)
    at async ESMLoader.import (node:internal/modules/esm/loader:527:24)
    at async loadESM (node:internal/process/esm_loader:91:5)
    at async handleMainPromise (node:internal/modules/run_main:65:12)
```
I always use `npx vite-node {script name}` to execute TypeScript files, but I'm not sure if we should recommend this way of executing to users. For me, it has always worked.

When I executed the script for the first time with this command, it took 44s to print the correct message.
While waiting, no output was printed to the console. So, if I'm a first time user, I'm not sure what's happening. Is the process doing something, did it hang?
I would add a `console.log()` before the execution to let the user know something is happening in the background and it should take some time to complete.

**Comment by crjm** [comment]
I found the issue, which seems to be `tsc`. For the record, I was using `npm` for this one.
By using the following command `tsc --init --module esnext --moduleResolution nodenext`, we are making the assumption that the user has `tsc` installed globally. When I execute that command I get:
```console
Command 'tsc' not found, but can be installed with:
sudo apt install node-typescript
```
Instead of making the user install the Typescript compiler with the OS package manager, we should use [`npx`](https://docs.npmjs.com/cli/v9/commands/npx).

This would avoid the user from installing tsc globally:
``````diff
- tsc --init --module esnext --moduleResolution nodenext
+ npx tsc --init --module esnext --moduleResolution nodenext
```

**Review by crjm — APPROVED** [review]
Works wonders! :ship:

**Comment by vikram-dagger** [comment]
@slumbering @crjm can you please suggest a fix for the lint errors in `build.js`?

**Comment by slumbering** [comment]
> @slumbering @crjm can you please suggest a fix for the lint errors in `build.js`?

@vikram-dagger : I've just modified the eslint configuration and set Node.js as a global environment. Waiting for the tests now.

File: docs/current/sdk/nodejs/snippets/get-started/step3/build.js [hunk 1]
```diff
@@ -2,29 +2,36 @@
...
-  }, {LogOutput: process.stdout})
+      // build application
+      // write the build output to the host
+      await runner
+        .withExec(["npm", "run", "build"])
+        .directory("build/")
+        .export("./build")
+      // highlight-end
+    },
+    { LogOutput: process.stdout }
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step3/build.js:35]: For consistency with other SDKs, we should use stderr :
```typescript
{ LogOutput: process.stderr }
```

File: docs/current/sdk/nodejs/snippets/get-started/step3/build.mjs [hunk 2]
```diff
@@ -1,28 +1,31 @@
...
-}, {LogOutput: process.stdout})
+    // build application
+    // write the build output to the host
+    await runner
+      .withExec(["npm", "run", "build"])
+      .directory("build/")
+      .export("./build")
+    // highlight-end
+  },
+  { LogOutput: process.stdout }
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step3/build.mjs:30]: For consistency with other SDKs, we should use stderr :
```typescript
{ LogOutput: process.stderr }
```

File: docs/current/sdk/nodejs/snippets/get-started/step3/build.mts [hunk 3]
```diff
@@ -1,28 +1,31 @@
...
-}, {LogOutput: process.stdout})
+    // build application
+    // write the build output to the host
+    await runner
+      .withExec(["npm", "run", "build"])
+      .directory("build/")
+      .export("./build")
+    // highlight-end
+  },
+  { LogOutput: process.stdout }
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step3/build.mts:30]: For consistency with other SDKs, we should use stderr :
```typescript
{ LogOutput: process.stderr }
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.js [hunk 4]
```diff
@@ -2,41 +2,46 @@
...
+        // build application using specified Node version
+        // write the build output to the host
+        await runner
+          .withExec(["npm", "run", "build"])
+          .directory("build/")
+          .export(`./build-node-${nodeVersion}`)
+      }
+      // highlight-end
+    },
+    { LogOutput: process.stdout }
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.js:45]: For consistency with other SDKs, we should use stderr :
```typescript
{ LogOutput: process.stderr }
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.mjs [hunk 5]
```diff
@@ -1,38 +1,43 @@
...
+      // build application using specified Node version
+      // write the build output to the host
+      await runner
+        .withExec(["npm", "run", "build"])
+        .directory("build/")
+        .export(`./build-node-${nodeVersion}`)
+    }
+    // highlight-end
+  },
+  { LogOutput: process.stdout }
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.mjs:42]: For consistency with other SDKs, we should use stderr :
```typescript
{ LogOutput: process.stderr }
```

File: docs/current/sdk/nodejs/snippets/get-started/step4/build.mts [hunk 6]
```diff
@@ -1,38 +1,43 @@
...
+      // build application using specified Node version
+      // write the build output to the host
+      await runner
+        .withExec(["npm", "run", "build"])
+        .directory("build/")
+        .export(`./build-node-${nodeVersion}`)
+    }
+    // highlight-end
+  },
+  { LogOutput: process.stdout }
```
Comment (slumbering) [inline docs/current/sdk/nodejs/snippets/get-started/step4/build.mts:42]: For consistency with other SDKs, we should use stderr :
```typescript
{ LogOutput: process.stderr }
```

**Review by slumbering — COMMENTED** [review]
Looks good to me. Just to keep a consistency, we should always use `process.stderr` now like in the other SDKs. Thanks

**Comment by vikram-dagger** [comment]
> Looks good to me. Just to keep a consistency, we should always use `process.stderr` now like in the other SDKs. Thanks

I would agree on doing this for other guides/pages (see https://github.com/dagger/dagger/issues/4858) but in the case of the get started _only_, I think it's important for the user to "see something happening" as it's their first experience with Dagger, as mentioned by @crjm in https://github.com/dagger/dagger/pull/5030/#pullrequestreview-1410904669. So I would be OK with this inconsistency, only for the get started guides for each SDK

**Pull request merged**